### PR TITLE
Feature/claude

### PR DESCRIPTION
**Pull request by Abraxas-365**

### CODE REVIEW

File: src/llm/cloude/client.rs [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    }
+
+    pub fn with_anthropic_version<S: Into<String>>(mut self, version: S) -> Self {
+        self.anthropic_version = version.into();
+        self
+    }
+
+    async fn generate(&self, messages: &[Message]) -> Result<GenerateResult, LLMError> {
+        let client = Client::new();
+        let is_stream = match &self.options.streaming_func {
```
Comment (prabirshrestha) [inline src/llm/cloude/client.rs:75]: probably can use `is_some()` or `is_none()` to inline into single line?

File: src/llm/cloude/client.rs [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            Some(_) => true,
+            None => false,
+        };
+
+        let payload = self.build_payload(messages, is_stream);
+        let res = client
+            .post("https://api.anthropic.com/v1/messages")
+            .header("x-api-key", &self.api_key)
+            .header("anthropic-version", self.anthropic_version.clone())
+            .header("content-type", "application/json")
```
Comment (prabirshrestha) [inline src/llm/cloude/client.rs:85]: Content-type: application/json; charset=utf-8

File: src/llm/cloude/client.rs [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        &self,
+        messages: &[Message],
+    ) -> Result<Pin<Box<dyn Stream<Item = Result<StreamData, LLMError>> + Send>>, LLMError> {
+        let client = Client::new();
+        let payload = self.build_payload(messages, true);
+        let request = client
+            .post("https://api.anthropic.com/v1/messages")
+            .header("x-api-key", &self.api_key)
+            .header("anthropic-version", &self.anthropic_version)
+            .header("content-type", "application/json")
```
Comment (prabirshrestha) [inline src/llm/cloude/client.rs:178]: Content-type: application/json; charset=utf-8

**Review by prabirshrestha — APPROVED** [review]

File: src/llm/cloude/client.rs (hunk 2 above)
Comment (Abraxas-365) [reply src/llm/cloude/client.rs]: Sorry for the ignorance, but why set charset=utf-8?

**Pull request merged**